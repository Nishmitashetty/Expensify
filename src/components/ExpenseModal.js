import React from 'react';
import Modal from 'react-modal';

const ExpenseModal = (props) =>
{
    return (
        <div>
           <Modal
           isOpen={!! props.selectedExpense}
           contentLabel="Selected Expense"
           closeTimeoutMS={200}
           className="modal"
           >
            <h1 className="modal__title">Delete</h1>
             <p className="modal__message">Are you sure do you want to delete this expense? </p>
             <div className="modal__buttons">
             <button 
             className="cancelButton"
             onClick={props.handleClearSelectedExpense}
             >Cancel
             </button>&nbsp;&nbsp;
             <button
             className="deleteButton"
             onClick={props.onRemove}
             >Delete
             </button>
             </div>
           </Modal>
        </div>
    );
};

export default ExpenseModal;