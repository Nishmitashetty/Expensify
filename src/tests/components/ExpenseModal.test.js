import React from 'react';
import { shallow } from 'enzyme';
import ExpenseModal from '../../components/ExpenseModal';

test('should render ExpenseModal component with default',() =>
{
    const onRemoveSpy = jest.fn();
    const handleClearSelectedExpense = jest.fn();

    const wrapper = shallow(
    <ExpenseModal
        selectedExpense={undefined}
        onRemove={onRemoveSpy}
        handleClearSelectedExpense={handleClearSelectedExpense}
    />);
    expect(wrapper).toMatchSnapshot();
});

let onRemove,handleClearSelectedExpense,wrapper;

beforeEach(() =>
{
    onRemove = jest.fn();
    handleClearSelectedExpense = jest.fn();

    wrapper = shallow(<ExpenseModal
        selectedExpense={'Rent'}
        onRemove={onRemove}
        handleClearSelectedExpense={handleClearSelectedExpense}
    />);
});

test('should render ExpenseModal component correctly',() =>
{
    expect(wrapper).toMatchSnapshot();
});

test('should handle cancel button',() =>
{
    wrapper.find('button').at(0).simulate('click');
    expect(handleClearSelectedExpense).toHaveBeenCalled();
});

test('should handle Delete button',() =>
{
    wrapper.find('button').at(1).simulate('click');
    expect(onRemove).toHaveBeenCalled();
});