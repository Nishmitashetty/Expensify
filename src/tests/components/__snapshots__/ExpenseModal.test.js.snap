// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseModal component correctly 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={200}
    contentLabel="Selected Expense"
    isOpen={true}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <h1
      className="modal__title"
    >
      Delete
    </h1>
    <p
      className="modal__message"
    >
      Are you sure do you want to delete this expense? 
    </p>
    <div
      className="modal__buttons"
    >
      <button
        className="cancelButton"
        onClick={[Function]}
      >
        Cancel
      </button>
        
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </Modal>
</div>
`;

exports[`should render ExpenseModal component with default 1`] = `
<div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal"
    closeTimeoutMS={200}
    contentLabel="Selected Expense"
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <h1
      className="modal__title"
    >
      Delete
    </h1>
    <p
      className="modal__message"
    >
      Are you sure do you want to delete this expense? 
    </p>
    <div
      className="modal__buttons"
    >
      <button
        className="cancelButton"
        onClick={[Function]}
      >
        Cancel
      </button>
        
      <button
        className="deleteButton"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </Modal>
</div>
`;
